kolibri/plugins/facility/assets/src/views/users/sidePanels/AssignCoachesSidePanel.vue

```diff
@@ -1,35 +1,347 @@
 <template>
 
-  <SidePanelModal
-    alignment="right"
-    sidePanelWidth="700px"
-    @closePanel="$router.back()"
-  >
-    <template #header>
-      <h1>{{ coreString('classesLabel') }}</h1>
-    </template>
-  </SidePanelModal>
+  <div>
+    <KModal
+      v-if="showUndoModal"
+      :title="
+        undoAssignCoachHeading$({
+          numUsers: eligibleUsersCount,
+          numClasses: selectedClasses.length,
+        })
+      "
+      :submitText="undoAction$()"
+      :cancelText="dismissAction$()"
+      :submitDisabled="isLoading"
+      :cancelDisabled="isLoading"
+      @cancel="handleDismissConfirmation"
+      @submit="handleUndoAssignments"
+    >
+      <KCircularLoader v-if="isLoading" />
+      <span
+        v-else
+        class="adjust-line-height"
+      >
+        {{
+          undoAssignCoachMessage$({
+            numUsers: eligibleUsersCount,
+            numClasses: selectedClasses.length,
+          })
+        }}
+      </span>
+    </KModal>
+    <SidePanelModal
+      v-else
+      alignment="right"
+      sidePanelWidth="700px"
+      :addBottomBorder="false"
+      @closePanel="closeSidePanel(selectedClasses.length > 0)"
```

Comment:
Just noting that here we should use the [useGoBack](https://github.com/lianaharris360/kolibri/blob/ccda0041b98f14fb0a7dd2f513b21252226ad48e/kolibri/plugins/facility/assets/src/views/users/sidePanels/FilterUsersSidePanel/index.vue#L131) composable to prevent the 4th point that Peter mentioned in this PR https://github.com/learningequality/kolibri/pull/13608#issuecomment-3167608877